### PR TITLE
Menu component

### PR DESCRIPTION
**Pull request by nicolobruno**
# Description

- Create Menu component generic with a button to set the width of the menu.
- Storybook `PENDING`

Example menuItems that it receive the Menu Component:

```
const menuItems = [
    {
      id: 1,
      name: 'HOME',
      path: '/home',
      icOn: iconOn,
      icOff: iconOff,
      subItems: [
        {
          id: 1,
          name: 'LOGIN',
          path: '/login'
        }
      ]
    },
    {
      id: 2,
      name: 'REPORTES',
      path: '/reportes',
      icOn: iconOn,
      icOff: iconOff,
      subItems: [
        {
          id: 1,
          name: 'Ventas',
          path: '/report/sales'
        },
        {
          id: 2,
          name: 'Inventario',
          path: '/report/stock'
        }
      ]
    }
  ];
```

## Screenshots 

menuboot

### CODE REVIEW

File: generators/app/templates/src/app/components/Menu/components/MenuItem/components/MenuSubItem/index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import MenuSubItem from './layout';
+import { subItemPropTypes } from './propTypes';
+
+class MenuSubItemContainer extends Component {
+  handleSubItemSelected = () => {
```
Comment (mmamaniwolox) [inline generators/app/templates/src/app/components/Menu/components/MenuItem/components/MenuSubItem/index.js]: Could be `handleSubItemSelected = () => this.props.onSubItemSelected(this.props.subItem);` ?

File: generators/app/templates/src/app/components/Menu/components/MenuItem/components/MenuSubItem/styles.module.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
+@import '../../../../../../../scss/variables/_colors';
```
Comment (mmamaniwolox) [inline generators/app/templates/src/app/components/Menu/components/MenuItem/components/MenuSubItem/styles.module.scss:1]: We do need relative imports :P

File: generators/app/templates/src/app/components/Menu/components/MenuItem/index.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import MenuItem from './layout';
+import { menuItemPropTypes } from './propTypes';
+
+class MenuItemContainer extends Component {
+  handleMenuItemSelected = () => {
```
Comment (mmamaniwolox) [inline generators/app/templates/src/app/components/Menu/components/MenuItem/index.js]: Could be `handleMenuItemSelected = () => this.props.handleItemSelected(this.props.menuItem);` ?

File: generators/app/templates/src/app/components/Menu/components/MenuItem/index.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    this.props.handleItemSelected(this.props.menuItem);
+  };
+
+  render() {
+    const isActive = this.props.currentSelectedItemId === this.props.menuItem.id;
```
Comment (mmamaniwolox) [inline generators/app/templates/src/app/components/Menu/components/MenuItem/index.js]: I think you could move this line below and use `currentSelectedItemId` and `menuItem` without `this.props`

File: generators/app/templates/src/app/components/Menu/components/MenuItem/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  overflow: hidden;
+  transition: max-height .25s ease-in-out;
+  background: $grey-0-1;
+
+  &.collapsed {
+    overflow-y: auto;
+    padding: 10px 0;
+    max-height: 150px;
+  }
+}
```
Comment (mmamaniwolox) [inline generators/app/templates/src/app/components/Menu/components/MenuItem/styles.module.scss]: Add a tailing line

File: generators/app/templates/src/app/components/Menu/index.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+MenuContainer.propTypes = {
+  handleItemSelected: PropTypes.func.isRequired,
+  handleSubItemSelected: PropTypes.func.isRequired,
+  menuItems: PropTypes.arrayOf(menuItemPropTypes).isRequired,
+  className: PropTypes.string
+};
+
+const mapDispatchToProps = dispatch => ({
+  handleItemSelected: selectedItem => {
+    if (window.location.pathname !== selectedItem.path) {
```
Comment (mmamaniwolox) [inline generators/app/templates/src/app/components/Menu/index.js]: Can this logic be moved to actions file? I haven't seen this in other projects

File: generators/app/templates/src/app/components/Menu/index.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      if (!selectedItem.subItems && selectedItem) {
+        dispatch(
+          push({
+            pathname: selectedItem.path
+          })
+        );
+      }
+    }
+  },
+  handleSubItemSelected: selectedSubItem => {
```
Comment (mmamaniwolox) [inline generators/app/templates/src/app/components/Menu/index.js]: Can be `handleSubItemSelected: selectedSubItem => dispatch(push({pathname: selectedSubItem.path})`?

File: generators/app/templates/src/app/components/Menu/components/MenuItem/components/MenuSubItem/styles.module.scss [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+.sub-item {
+  padding: 10px;
```
Comment (FrankIglesias) [inline generators/app/templates/src/app/components/Menu/components/MenuItem/components/MenuSubItem/styles.module.scss]: use shorthand for paddings

File: generators/app/templates/src/app/components/Menu/styles.module.scss [hunk 9]
```diff
@@ -0,0 +1,26 @@
+@import '../../../scss/variables/_colors';
+@import '../../../scss/variables/_sizes';
+
+.menu {
+  width: $sidebar-width;
+  padding-top: 30px;
+  box-shadow: 2px 0 4px 0 rgba($black, 0.25);
```
Comment (FrankIglesias) [inline generators/app/templates/src/app/components/Menu/styles.module.scss]: use color variable

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: generators/app/templates/src/app/components/Menu/propTypes.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import PropTypes from 'prop-types';
+
+export const menuItemPropTypes = PropTypes.shape({
```
Comment (briangerez) [inline generators/app/templates/src/app/components/Menu/propTypes.js]: There is already a menuItemPropTypes in `src / app / components / Menu / components / MenuItem / propTypes.js` ?

**Pull request closed without merging**